File: commonUI/src/main/java/com/ionos/common_ui/dialog/wrapper/NoFreeSpaceMessageDialogWrapper.java
package com.ionos.common_ui.dialog.wrapper;

import android.content.Context;

import com.ionos.common_ui.dialog.stylized.localized.StringResLocalizedStrategy;
import com.ionos.common_ui.R;
import com.ionos.common_ui.di.CommonUiComponent;
import com.ionos.common_ui.dialog.stylized.StylizedDialog;
import com.ionos.common_ui.utils.ContextUtils;
import com.ionos.common_ui.exception.TryCatchExceptionHandler;

import javax.inject.Inject;

/**
 * Created by Sergey Shandyuk on 12/30/2016.
 */

public class NoFreeSpaceMessageDialogWrapper {
    //TODO alk
//	@Inject
//	ICustomFonts customFonts;
	@Inject
	TryCatchExceptionHandler tryCatchExceptionHandler;

	private final StylizedDialog dialog;

	public NoFreeSpaceMessageDialogWrapper(Context context) {
		CommonUiComponent.from(context).inject(this);

		this.dialog = StylizedDialog.newBuilder(this.tryCatchExceptionHandler)
				.setTitle(new StringResLocalizedStrategy(context, R.string.no_enough_free_device_space_title))
				.setMessage(new StringResLocalizedStrategy(context, R.string.scanbot_no_enough_free_device_space_message))
				.setPositiveButton(new StringResLocalizedStrategy(context, R.string.ok_btn_title), StylizedDialog::dismiss)
				.setButtonsTextColor(R.color.stylized_edit_text_line_color)
//				.setTypeface(this.customFonts.getMainRegularFont())
				.build(context);
	}

	public final void show(Context context) {
		if (!ContextUtils.isActivityFinishing(context)) {
			this.dialog.show();
		}
	}
}

File: scanbot/src/main/java/com/ionos/scanbot/screens/camera/ui_components/NoFreeSpaceMessageDialogWrapper.kt
package com.ionos.scanbot.screens.camera.ui_components

import android.content.Context
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.ionos.common_ui.utils.ContextUtils
import com.ionos.scanbot.R
import com.ionos.scanbot.di.inject
import kotlin.getValue

class NoFreeSpaceMessageDialogWrapper(context: Context) {
    private val tryCatchExceptionHandler by context.inject { tryCatchExceptionHandler() }

    private val dialog by lazy {
        MaterialAlertDialogBuilder(context)
            .setTitle(R.string.no_enough_free_device_space_title)
            .setMessage(R.string.scanbot_no_enough_free_device_space_message)
            .setPositiveButton(R.string.ok_btn_title) { dialog, _ -> dialog.dismiss() }
            .create()
    }

    fun show(context: Context?) {
        if (!ContextUtils.isActivityFinishing(context)) {
            tryCatchExceptionHandler.handle(dialog::show)
        }
    }
}
